instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -179,7 +180,8 @@ def __init__(self, *args, **kwargs):
         self.wsgi_app = _rewrapped_app(self.wsgi_app)
 
         _before_request = _wrapped_before_request(
-            _InstrumentedFlask.name_callback
+            _InstrumentedFlask.name_callback,
+            tracer_provider=_InstrumentedFlask._tracer_provider,
```

Comment:
How about passing in an instance of a tracer to this function instead of a provider. That way it won't have to call get_tracer for every incoming request.

Reply:
Yeah, I was looking into this. Few instrumentation have this problem and others have followed different ways to avoid creating tracer instance on each call. It should be achievable here as well. I am fine with just passing tracer as long as we keep it consistent across instrumentations. WDYT?

Reply:
Updated this to create one tracer instance when instrumentation is enabled and use it for all.